Sources/Sentry/SentrySessionTaskSearch.m

```diff
@@ -0,0 +1,57 @@
+/**
+ * Part of this code was copied from
+ * https://github.com/AFNetworking/AFNetworking/blob/4eaec5b586ddd897ebeda896e332a62a9fdab818/AFNetworking/AFURLSessionManager.m#L407-L418
+ * under the MIT license
+ */
+
+#import "SentrySessionTaskSearch.h"
+#import <objc/runtime.h>
+
+@implementation SentrySessionTaskSearch
+
++ (NSArray<Class> *)urkSessionTaskClassesToTrack
+{
+
+    /**
+     * In order to be able to track a network request, we need to know when it starts and when it
+     * finishes. NSURLSessionTask has a `resume` method that starts the request, and the only way to
+     * know when it finishes is to check the task `state`. Using KVO is not working, so we are
+     * swizzling `setState:`. Depending on the iOS version NSURLSessionTask does not implements
+     * `setState:` and Apple uses a subclass returned by NSURLSession that implementes `setState:`.
+     * We need to discover which class to swizzle.
+     *
+     * Apple intermediate class for iOS does not call [super resume], so we can swizzle both class.
```

Comment:
`m`: I think it would be nice to point out the detailed comment in AFNewtorking https://github.com/AFNetworking/AFNetworking/blob/4eaec5b586ddd897ebeda896e332a62a9fdab818/AFNetworking/AFURLSessionManager.m#L356-L381. We could even copy it in here.